templates/facetDemoScript.template.hbs

```diff
@@ -0,0 +1,165 @@
+    transport: http(RPC_URL),
+    account: walletAccount,
+  })
+  const signerAddress = walletAccount.address
+
+  // === Contract addresses ===
+  // LIFI_ADDRESS is the address of the LiFi Diamond contract for bridging.
+  // SRC_TOKEN_ADDRESS is the token to be bridged (set its address for the source chain).
+  // EXAMPLE_ADDRESS is a custom contract address for a contract that your facet interacts with.
+  const LIFI_ADDRESS = deployments.LiFiDiamond as `0x${string}` // Set correct LiFiDiamond address
+  const SRC_TOKEN_ADDRESS = '' as `0x${string}` // Set the source token address here.
+  const EXAMPLE_ADDRESS = config.mainnet.example as `0x${string}` // Update for the correct chain.
+
+  // === Instantiate contracts ===
+  const srcTokenContract = getContract({
+    address: SRC_TOKEN_ADDRESS,
+    abi: ERC20_ABI,
+    client: { public: publicClient, wallet: walletClient }
+  })
+
+  const exampleContract = getContract({
+    address: EXAMPLE_ADDRESS,
+    abi: EXAMPLE_ABI,
+    client: { public: publicClient, wallet: walletClient }
+  })
+
+  const lifiDiamondContract = getContract({
+    address: LIFI_ADDRESS,
+    abi: {{constantCase name}}_FACET_ABI,
+    client: { public: publicClient, wallet: walletClient }
+  })
+
+  const srcTokenName = await srcTokenContract.read.name() as string
+  const srcTokenSymbol = await srcTokenContract.read.symbol() as string
+  const amount = parseUnits('10', 18) // 10 * 1e18
+
+  console.info(`\Bridge ${amount} ${srcTokenName} (${srcTokenSymbol}) from ${srcChain} --> {DESTINATION CHAIN NAME}`) // Set destination chain name
+  console.info(`Connected wallet address: ${signerAddress}`)
+
+  // === Check token balance ===
+  const balance = await srcTokenContract.read.balanceOf([signerAddress])
+  console.info(`Token balance for connected wallet: ${balance}`)
+
+  if (balance === 0n) {
+    console.error(`Connected account has no ${srcTokenName} (${srcTokenSymbol}) tokens. Exiting...`)
+    process.exit(1)
+  }
+
+  // === Check allowance ===
+  const currentAllowance = await srcTokenContract.read.allowance([
+    signerAddress,
+    LIFI_ADDRESS,
+  ]) as bigint;
+
+  console.log('Current allowance:', currentAllowance.toString())
+  
+  if (currentAllowance < amount) {
+    console.info('Allowance is insufficient. Approving required amount...')
+    try {
+      const hash = await srcTokenContract.write.approve([LIFI_ADDRESS, amount])
+      console.info(`Approval transaction sent (hash): ${hash}`)
+    } catch (error) {
+      console.error('Approval failed:', error)
+      process.exit(1)
+    }
+    console.info('Approval transaction complete. New allowance set.')
+  } else {
+    console.info('Sufficient allowance already exists. No need to approve.')
+  }
+
+  // === In this part put necessary logic usually it's fetching quotes, estimating fees, signing messages etc. ===
+
+
+
+
+  // === Prepare bridge data ===
+  // Generate types by running command `yarn typechain`
```

Comment:
wondering if it makes sense to have some sort of helper function that checks if the types for this particular facet exist and if not, it will run the command to make sure that the required types are available

Reply:
ok, that's fine